OData/test/E2ETest/WebStack.QA.Test.OData/AlternateKeys/AlternateKeysTest.cs

```diff
@@ -0,0 +1,334 @@
+            response = await Client.GetAsync(requestUri);
+            response.EnsureSuccessStatusCode();
+            string guidRespoonse = await response.Content.ReadAsStringAsync();
+
+            Assert.Equal(primitiveResponse, nameResponse);
+            Assert.Equal(primitiveResponse, guidRespoonse);
+        }
+
+        [Fact]
+        public async Task QueryEntityWithComposedAlternateKeys_Returns_SameEntityWithPrimitiveKey()
+        {
+            // query with declared key
+            var requestUri = string.Format("{0}/odata/People(2)", this.BaseAddress);
+            HttpResponseMessage response = await Client.GetAsync(requestUri);
+
+            response.EnsureSuccessStatusCode();
+            string primitiveResponse = await response.Content.ReadAsStringAsync();
+
+            // query with composed alternate keys
+            requestUri = string.Format("{0}/odata/People(Country='United States',Passport='9999')", this.BaseAddress);
+            response = await Client.GetAsync(requestUri);
+            response.EnsureSuccessStatusCode();
+            string composedResponse = await response.Content.ReadAsStringAsync();
+
+            Assert.Equal(primitiveResponse, composedResponse);
+        }
+
+        [Fact]
+        public async Task QueryFailedIfMissingAnyOfComposedAlternateKeys()
+        {
+            var requestUri = string.Format("{0}/odata/People(Country='United States')", this.BaseAddress);
+            var response = await Client.GetAsync(requestUri);
+            Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+        }
+
+        /* Not supported now: see github issue: https://github.com/OData/odata.net/issues/294
+         * if supported. modify the following test
+        [Fact]
+        public async Task QueryEntityWithComplexPropertyAlternateKeys_Returns_SameEntityWithPrimitiveKey()
+        {
+            var requestUri = string.Format("{0}/odata/Companies(2)", this.BaseAddress);
+            HttpResponseMessage response = await Client.GetAsync(requestUri);
+
+            Console.WriteLine(response.StatusCode);
+            string responseContent = await response.Content.ReadAsStringAsync();
+            Console.WriteLine(requestUri);
+            Console.WriteLine(responseContent);
+
+            requestUri = string.Format("{0}/odata/People(Country='United States',Passport='9999')", this.BaseAddress);
+            response = await Client.GetAsync(requestUri);
+            responseContent = await response.Content.ReadAsStringAsync();
+            Console.WriteLine(requestUri);
+            Console.WriteLine(responseContent);
+        }
+         * */
+
+        [Fact]
+        public async Task CanUpdateEntityWithSingleAlternateKeys()
+        {
+            const string expect = @"{
+  ""@odata.context"":""http://sam-xu-odata:9001/odata/$metadata#Customers/$entity"",""ID"":6,""Name"":""Updated Customer Name"",""SSN"":""SSN-6-T-006""
```

Comment:
You need to remove the odata.context which will cause it to fail on another machine.